pkg/apis/core/types.go

```diff
@@ -2635,6 +2650,34 @@ const (
 	PodQOSBestEffort PodQOSClass = "BestEffort"
 )
 
+// PodDNSConfig defines the DNS parameters of a pod in addition to
+// those generated from DNSPolicy.
+type PodDNSConfig struct {
+	// A list of DNS name server IP addresses.
+	// This will be appended to the base nameservers generated from DNSPolicy.
+	// Duplicated nameservers will be removed.
+	// +optional
+	Nameservers []string
+	// A list of DNS search domains for host-name lookup.
+	// This will be appended to the base search paths generated from DNSPolicy.
+	// Duplicated search paths will be removed.
+	// +optional
+	Searches []string
+	// A list of DNS resolver options.
+	// This will be merged with the base options generated from DNSPolicy.
+	// If duplicated, options given by this will overwrite the base options.
```

Comment:
Duplicated entries will be removed. Resolution options given in `Options` will override those that appear in the base DNSPolicy.

Reply:
Thanks, applied.